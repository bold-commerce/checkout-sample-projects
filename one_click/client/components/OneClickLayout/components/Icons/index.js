export { default as TimesCircle } from './TimesCirlce'
export { default as ArrowRight } from './ArrowRight'
export { default as CaretRight } from './CaretRight'