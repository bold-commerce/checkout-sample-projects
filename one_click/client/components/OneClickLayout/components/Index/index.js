export { default as IndexPage } from "./IndexPage";
export { default as IndexGuest } from "./IndexGuest";