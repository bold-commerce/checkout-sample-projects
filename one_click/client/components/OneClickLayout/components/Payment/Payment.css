.FieldSet--PaymentMethod .FieldSet__Content iframe {
  width: 100%;
  height: 100%;
  border: none;
}

.FieldSet--PaymentMethod .FieldSet__Content iframe .ToggleField__Input:checked {
  box-shadow: 0 0 0 5px #0099cc inset;
}

section.FieldSet.FieldSet--PaymentMethod {
  margin: 0;
}

section.FieldSet.FieldSet--PaymentMethod .FieldSet__Content {
  margin-bottom: 16px;
}
