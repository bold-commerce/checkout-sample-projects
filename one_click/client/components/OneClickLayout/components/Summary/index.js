export { default as Summary } from './Summary';
export { default as SummaryCondensed } from './SummaryCondensed';