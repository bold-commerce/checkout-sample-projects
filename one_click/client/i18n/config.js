import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;