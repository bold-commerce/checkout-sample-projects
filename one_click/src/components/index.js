export * from './Address';
export * from './BackButton';
export * from './Billing';
export * from './BillingAddress';
export * from './Card';
export * from './CheckoutButton';
export * from './CheckoutForm';
export * from './Confirmation';
export * from './CustomerInfo';
export * from './Discount';
export * from './EmptyState';
export * from './Header';
export * from './Icons';
export * from './Index';
export * from './InventoryItem';
export * from './LineItems';
export * from './LoadingState';
export * from './Payment';
export * from '../pages/Processing';
export * from './Product';
export * from './RadioItem';
export * from './Shipping';
export * from './ShippingLines';
export * from './Summary';
