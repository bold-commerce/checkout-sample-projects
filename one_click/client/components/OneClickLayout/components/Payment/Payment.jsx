import React from 'react';
import PropTypes from 'prop-types';
import EmptyState from '../EmptyState/EmptyState';
import { BackButton } from '../BackButton';
import { usePaymentMethod } from '@boldcommerce/checkout-react-components';
import PaymentIframe from './PaymentIframe';
import './Payment.css';

export const PaymentMethod = ({ showPaymentMethod }) => {
  if (showPaymentMethod) {
    return (
      <div className="Payment">
        <BackButton />
        <section className="FieldSet FieldSet--PaymentMethod">
          <div className="FieldSet__Header">
            <div className="FieldSet__Heading">Payment methods</div>
          </div>
          <div className="FieldSet__Content">
            <EmptyState title="To view payment options, complete filling in your address" />
            <PaymentIframe key="paymentIframe" hide />
          </div>
        </section>
      </div>
    );
  }

  return (
    <section className="FieldSet FieldSet--PaymentMethod">
      <div className="FieldSet__Header">
        <div className="FieldSet__Heading">Payment methods</div>
      </div>
      <div className="FieldSet__Content">
        <PaymentIframe key="paymentIframe" />
      </div>
    </section>
  );
};

PaymentMethod.propTypes = {
  showPaymentMethod: PropTypes.bool,
};

const MemoizedPaymentMethod = React.memo(PaymentMethod);

const PaymentMethodContainer = () => {
  const { showPaymentMethod } = usePaymentMethod();

  return <MemoizedPaymentMethod showPaymentMethod={showPaymentMethod} />;
};

export default PaymentMethodContainer;
