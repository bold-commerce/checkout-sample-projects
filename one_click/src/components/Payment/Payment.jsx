import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import EmptyState from '../EmptyState/EmptyState';
import PaymentIframe from './PaymentIframe';
import { Discount } from '../Discount';
import { useLocation } from 'react-router-dom';
import { BillingAddress } from '../BillingAddress';
import { useBillingAddress, useShippingLines } from '@boldcommerce/checkout-react-components';
import './Payment.css';
import { useTranslation } from 'react-i18next';

const PaymentMethod = ({ billingAddress, shippingLines }) => {
  const location = useLocation();
  const hidePaymentComponent = !(location.pathname === "/");
  const disabled = Array.isArray(billingAddress) && shippingLines.length === 0;
  const { t } = useTranslation();

  if (disabled) {
    return (
      <div className='Payment'>
        <section className={classnames(
          {'hidden': hidePaymentComponent ,
          'FieldSet FieldSet--PaymentMethod': true
        })}>
          <div className="FieldSet__Content">
            <EmptyState title={t('payment.empty')} />
            <PaymentIframe key="paymentIframe" hide />
          </div>
        </section>
      </div>
    );
  }

  return (
    <>
      <div className='Payment'>
        <section className={classnames(
          {'hidden': hidePaymentComponent,
          'FieldSet FieldSet--PaymentMethod': true
        })}>
          <div className="FieldSet__Content">
            <PaymentIframe key="paymentIframe" />
          </div>
          <div className="FieldSet__Content">
            <Discount />
          </div>
          <div className="FieldSet__Content">
            <BillingAddress />
          </div>
        </section>
      </div>
    </>
  );
};

PaymentMethod.propTypes = {
  showPaymentMethod: PropTypes.bool,
};

const MemoizedPaymentMethod = React.memo(PaymentMethod);

const PaymentMethodContainer = () => {
  const { data: billingAddress } = useBillingAddress();
  const { data } = useShippingLines(); 
  const { shippingLines } = data;

  return <MemoizedPaymentMethod shippingLines={shippingLines} billingAddress={billingAddress} />;
};

export default PaymentMethodContainer;
