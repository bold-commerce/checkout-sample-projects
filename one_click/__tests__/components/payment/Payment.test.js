import React from "react";
import { render } from '@testing-library/react';
import { PaymentMethod } from '../../../src/components/Payment'
import { MemoryRouter } from "react-router";
import { exampleShippingState as  MOCKexampleShippingState } from '../../utils/shippingLinesHelper';
import { 
    countries as MOCKcountries,
    caProvinces as MOCKprovinces,
    exampleAddress as MOCKexampleAddress
} from '../../utils/addressHelpers';
import '../../../src/i18n/config';

jest.mock('@boldcommerce/checkout-react-components', () => ({
    ...jest.requireActual('@boldcommerce/checkout-react-components'),
    usePaymentIframe: () => ({
        data: {
            url: 'test.url',
            loadingStatus: 'fulfilled',
            paymentIframeOnLoaded: (() => {})
        }
    }),
    useCountryInfo: () => ({
        data: {
            countries: MOCKcountries,
            provinceLabel: "province",
            provinces: MOCKprovinces,
            showPostalCode: true,
            showProvince: true
        }
    }),
    useDiscount: () => ({
        data: {
            discountApplied: false,
            discountCode: "",
            errors: null,
            applyDiscount: (() => {})
        }
    }),
    useBillingAddress: () => ({
        billingAddress: MOCKexampleAddress,
        countryInfo: MOCKcountries,
        billingAddressErrors: null,
        billingSameAsShipping: true,
        submitBillingAddress: (() => {}),
        setBillingSameAsShipping: (() => {})
    }),
    useShippingLines: () => ({
        data: MOCKexampleShippingState
    }),
    useBillingSameAsShipping: () => ({
        data: true,
        setBillingSameAsShipping: (() => {})
    })
}))

describe('Payment', () => {
    test('renders PaymentMethod component showing payment method', () => {
        const { asFragment } = render(
            <PaymentMethod showPaymentMethod />,
            {wrapper: MemoryRouter} );
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders PaymentMethod component without showing payment method', () => {
        const { asFragment } = render(
            <PaymentMethod showPaymentMethod={false} />,
                {wrapper: MemoryRouter}
             );
        expect(asFragment()).toMatchSnapshot();
    });
});