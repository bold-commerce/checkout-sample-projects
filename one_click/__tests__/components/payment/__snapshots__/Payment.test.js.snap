// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment renders PaymentMethod component showing payment method 1`] = `
<DocumentFragment>
  <div
    class="Payment"
  >
    <section
      class="hidden FieldSet FieldSet--PaymentMethod"
    >
      <div
        class="FieldSet__Content"
      >
        <iframe
          data-bold-pigi-iframe="true"
          src="test.url"
          title="payments"
        />
      </div>
      <div
        class="FieldSet__Content"
      >
        <div
          class="DiscountLink"
        >
          Discount code
        </div>
      </div>
      <div
        class="FieldSet__Content"
      >
        <section
          class="FieldSet FieldSet--BillingAddress"
        >
          <div
            class="FieldSet__Header"
          >
            <h3>
              Billing address
            </h3>
          </div>
          <div
            class="FieldSet__Content"
          >
            <div
              class="stx-field stx-checkbox-field__field RadioButton"
            >
              <div
                class="stx-checkbox-field__div"
              >
                <span
                  class="stx-checkbox stx-checkbox--is-checked stx-checkbox-field__checkbox"
                >
                  <input
                    checked=""
                    class="stx-checkbox__input"
                    id="stxField1"
                    name="billing-address"
                    type="checkbox"
                    value="SAME_AS_SHIPPING_ADDRESS"
                  />
                  <span
                    class="stx-checkbox__mount"
                  />
                </span>
                <label
                  class="stx-checkbox-field__label"
                  for="stxField1"
                >
                  Same as shipping address
                </label>
              </div>
            </div>
          </div>
        </section>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Payment renders PaymentMethod component without showing payment method 1`] = `
<DocumentFragment>
  <div
    class="Payment"
  >
    <section
      class="hidden FieldSet FieldSet--PaymentMethod"
    >
      <div
        class="FieldSet__Content"
      >
        <iframe
          data-bold-pigi-iframe="true"
          src="test.url"
          title="payments"
        />
      </div>
      <div
        class="FieldSet__Content"
      >
        <div
          class="DiscountLink"
        >
          Discount code
        </div>
      </div>
      <div
        class="FieldSet__Content"
      >
        <section
          class="FieldSet FieldSet--BillingAddress"
        >
          <div
            class="FieldSet__Header"
          >
            <h3>
              Billing address
            </h3>
          </div>
          <div
            class="FieldSet__Content"
          >
            <div
              class="stx-field stx-checkbox-field__field RadioButton"
            >
              <div
                class="stx-checkbox-field__div"
              >
                <span
                  class="stx-checkbox stx-checkbox--is-checked stx-checkbox-field__checkbox"
                >
                  <input
                    checked=""
                    class="stx-checkbox__input"
                    id="stxField2"
                    name="billing-address"
                    type="checkbox"
                    value="SAME_AS_SHIPPING_ADDRESS"
                  />
                  <span
                    class="stx-checkbox__mount"
                  />
                </span>
                <label
                  class="stx-checkbox-field__label"
                  for="stxField2"
                >
                  Same as shipping address
                </label>
              </div>
            </div>
          </div>
        </section>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
